lib/chef/knife/vsphere_vm_state.rb

```diff
@@ -110,6 +110,9 @@ def run
       when 'reset'
         vm.ResetVM_Task.wait_for_completion
         puts "Reset virtual machine #{vmname}"
+      when 'reboot'
+        vm.RebootGuest
```

Comment:
Unfortunately, this method doesn' t return a task object.

From [API documentation](https://www.vmware.com/support/developer/converter-sdk/conv50_apireference/vim.VirtualMachine.html#rebootGuest):
>RebootGuest
>Issues a command to the guest operating system asking it to perform a reboot. Returns immediately and does not wait for the guest operating system to complete the operation.

I tried to implement a wait option, but I didn't have any idea how to achieve this - I think VMware didn't as well.